pkgs/tools/networking/bandwhich/default.nix

```
@@ -23,11 +23,25 @@ rustPlatform.buildRustPackage rec {
     "--skip=tests::cases::ui::layout_under_50_width_under_50_height"
   ];
 
+  nativeBuildInputs = [ installShellFiles ];
+
   buildInputs = lib.optional stdenv.isDarwin Security;
 
   # 10 passed; 47 failed https://hydra.nixos.org/build/148943783/nixlog/1
   doCheck = !stdenv.isDarwin;
 
+  preConfigure = ''
+    export BANDWHICH_GEN_DIR=_shell-files
+    mkdir -p $BANDWHICH_GEN_DIR
+  '';
+
+  postInstall = ''
+    installManPage $BANDWHICH_GEN_DIR/bandwhich.1
+
+    installShellCompletion $BANDWHICH_GEN_DIR/bandwhich.{bash,fish} \
```

Does this work like it should? I would expect the below to work reliable.


```suggestion
    installShellCompletion --cmd bandwhich \
      --bash $BANDWHICH_GEN_DIR/bandwhich.bash \
      --fish $BANDWHICH_GEN_DIR/bandwhich.fish \
```

---

`installShellCompletion` has a bunch of auto-detection logic[^1].

This essentially boils down to:
- `--cmd` isn't needed, because the completions already contain the command (e.g. `bandwhich` in `bandwhich.bash`)
- `bandwhich.bash` and `bandwhich.fish` have proper file extensions (`.bash` and `.fish`) from which the shell is automatically inferred[^2]
- `_bandwhich` for zsh is the only completion that needs special care, unless one wants an ugly `installShellCompletion: warning: assuming path $arg is zsh; please specify with --zsh` warning  [^3]

But I can use your version/variant, if you prefer it that way. No hard feelings.

[^1]: https://github.com/NixOS/nixpkgs/blob/55ad4874cf42a9c1425840a9907a68234b74945a/pkgs/build-support/setup-hooks/install-shell-files.sh#L52-L97

[^2]: https://github.com/NixOS/nixpkgs/blob/55ad4874cf42a9c1425840a9907a68234b74945a/pkgs/build-support/setup-hooks/install-shell-files.sh#L162-L167

[^3]: https://github.com/NixOS/nixpkgs/blob/55ad4874cf42a9c1425840a9907a68234b74945a/pkgs/build-support/setup-hooks/install-shell-files.sh#L168-L172

---

@SuperSandro2000 and @figsoda are you fine with me marking this as resolved?

---

fine for me